### DISCUSSION TITLE
List component

### DISCUSSION THREAD

**Discussion by mslizh** [chakra-ui/zag#1495] [Ideas]
# 🚀 Feature request

> Please describe your request in one or two sentences.

Add a component for a list that allows keyboard navigating, adding, deleting, selecting (one or more) and dragging and dropping items. Some of the functionality should be similar to the way lists work in Combobox, Select, Menu, etc.

## 🧱 Problem Statement / Justification

> Please provide valid reason(s) why this should added.
>
> If this feaure is related to a problem you've noticed. Mention it as well

Because it's a popular functionality in many applications that has to be constantly re-implemented. Keyboard functionality are often forgotten.

## ✅ Proposed solution or API

> Please provide code snippets, gists, or links to the ideal design or API

Sorry, I'm a beginner so I don't have any.

## ↩️ Alternatives

> What alternative solutions have you considered before making this request?

[React Aria List Box](https://react-spectrum.adobe.com/react-aria/ListBox.html)

## 📝 Additional Information

> What resources (links, screenshots, etc.) do you have to assist this effort?

**Comment by anubra266** [comment]
@mslizh If you need a list, you can use the select machine.

```js
const [state, send] = useMachine(
  select.machine({
    // ...
    open: true,
    "open.controlled": true,
  })
);

const api = select.connect(state, send, normalizeProps);

return (
  <div {...api.rootProps}>
    <ul {...api.contentProps} style={{ height: "300px" }}>
      {selectData.map((item) => (
        <li key={item.value} {...api.getItemProps({ item })}>
          <span {...api.getItemTextProps({ item })}>{item.label}</span>
          <span {...api.getItemIndicatorProps({ item })}>✓</span>
        </li>
      ))}
    </ul>
  </div>
);
```


https://github.com/chakra-ui/zag/assets/30869823/e48581e5-708a-4251-89f2-6296302fb776

cc @segunadebayo

**Comment by segunadebayo** [comment]
Closing — a `@zag-js/listbox` machine now exists.
